refactor: material spec 2

iCrawl opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/valentingavran/anglify/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

iCrawl commented:
List of (maybe) controversial changes in this PR:

- Removal of `elevated` button style.
- - There is no reason for us to do this, we can already use the elevation utility for this.
- Rename of `filled` button style to `contained` as per material spec.
- Removal of `filled-tonal` button style.
- - We can already handle changing the button color and text color within just a certain button really easily, no need to do this.
- Removal of `sizing` property on buttons.
- - We can handle changing the button size with CSS rather easily too, no need to do all out here and change it weirdly with props to have 5 fixed sizes.
- Removal of `sizing` property on fab buttons.
- - See above.

All other changes are just to conform with material spec and avoid a bazillion colors to be configured without knowing where and how they get used. We had quite some css vars that were simply not even used lol.

valentingavran reviewed (changes requested):
Thanks for the PR. The changes will be a great addition. 🚀

However, there are still a few things that need to be tweaked. Only then I can merge the PR

- `--color-surface-overlay-darker` is used for the stepper but it is not defined in the light theme. Please add this or use an alternative
- Please create another PR where you do the code formatting (removing return types, adding missing lines, ...). This makes the commit history more readable.
- Please add a footer to this commit, where you describe the breaking changes. This is explained in more detail in the [CONTRIBUTING.md](https://github.com/valentingavran/anglify/blob/master/CONTRIBUTING.md) in the lines 267-278

Edit 1:
- Icon buttons are also no longer rounded

Edit 2:
- Please add again two separate CSS custom properties to distinguish between hover and focus colors

![image](https://user-images.githubusercontent.com/28765584/167265241-578c9121-d4f7-4cb5-addc-e130f36be68f.png)
Source: https://material.io/design/interaction/states.html#anatomy

iCrawl commented:
- Removed `--color-surface-overlay-dark` and replaced it with the normal `--color-surface-overlay` for now.
- - More color changes need to be revisited in the dark theme PR anyway, so its a good point to tackle it there again.
- Removed all other changes, will be done in a separate PR.
- Added footer message for breaking changes.
- Icon buttons are rounded now.
- There already are 2 CSS variables for those states, namely: `--color-primary-state-hover` and `--color-primary-state-focus`
